File: frontend/src/types/address.ts
/**
 * Address Types - Matching Backend API
 */

export type AddressType = 'SHIPPING' | 'BILLING';

export interface Address {
  id: number;
  addressLine1: string;
  addressLine2?: string;
  city: string;
  stateProvince: string;
  postalCode: string;
  country: string;
  phoneNumber?: string;
  type: AddressType;
  isDefault: boolean;
}

export interface CreateAddressRequest {
  addressLine1: string;
  addressLine2?: string;
  city: string;
  stateProvince: string;
  postalCode: string;
  country: string;
  phoneNumber?: string;
  type: AddressType;
  isDefault?: boolean;
}

export interface UpdateAddressRequest {
  addressLine1: string;
  addressLine2?: string;
  city: string;
  stateProvince: string;
  postalCode: string;
  country: string;
  phoneNumber?: string;
  type: AddressType;
  isDefault?: boolean;
}

export interface AddressListResponse {
  addresses: Address[];
  count: number;
}

export interface AddressResponse {
  message: string;
  address: Address;
}

File: backend/src/main/java/com/ut/edu/backend/user/Address.java
package com.ut.edu.backend.user;

import com.ut.edu.backend.common.BaseEntity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

/**
 * Address entity for user shipping and billing addresses
 */
@Entity
@Table(name = "addresses")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(callSuper = true, exclude = "user")
public class Address extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;

    @NotBlank(message = "Address line 1 is required")
    @Column(nullable = false)
    private String addressLine1;

    private String addressLine2;

    @NotBlank(message = "City is required")
    @Column(nullable = false, length = 100)
    private String city;

    @NotBlank(message = "State/Province is required")
    @Column(nullable = false, length = 100)
    private String stateProvince;

    @NotBlank(message = "Postal code is required")
    @Column(nullable = false, length = 20)
    private String postalCode;

    @NotBlank(message = "Country is required")
    @Column(nullable = false, length = 100)
    private String country;

    @Column(nullable = false)
    @Builder.Default
    private Boolean isDefault = false;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private AddressType type = AddressType.SHIPPING;
}
